src/librustdoc/clean/mod.rs

```diff
@@ -2171,7 +2171,8 @@ impl Clean<Vec<Item>> for doctree::Import<'_> {
                         }
                         None => false,
                     }
-            });
+            })
+            || (self.vis.node.is_pub() && self.name == kw::Underscore);
```

Comment:
Also, I'm not sure this logic is correct - it prevents you from adding `doc(inline)` on the trait. But maybe that's the expected behavior, since there's no good behavior documentation when it's inlined? In that case I'd expect an error instead of silently ignoring the attribute. @Nemo157 what do you think?

(Related: https://github.com/rust-lang/rust/issues/80275)

Reply:
An error for `doc(inline)` sounds fine, it would be possible to have a _single_ `#[doc(inline)] pub use something as _;` that would avoid the overlapping files, but I don't think it's useful enough to support.

Reply:
I changed to line 2165 as requested.
As for the error, what should be used? ```rustc_errors``` or ```panic!``` or other? This verification should be done before the ```let mut denied = ... ``` ?
Also what wording should be used in the message?

Reply:
See https://rustc-dev-guide.rust-lang.org/diagnostics.html, you'd add a new [error code](https://rustc-dev-guide.rust-lang.org/diagnostics/diagnostic-codes.html) and use it with [`struct_span_err!`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/macro.struct_span_err.html). See https://github.com/rust-lang/rust/blob/75e1acb63a642fa78cb8c382d7462b41fa2b56cd/compiler/rustc_attr/src/builtin.rs#L33 for an example.